sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinals.scala

```diff
@@ -42,13 +42,5 @@ class SubstituteUnresolvedOrdinals(conf: SQLConf) extends Rule[LogicalPlan] {
         case other => other
       }
       withOrigin(s.origin)(s.copy(order = newOrders))
-
-    case a: Aggregate if conf.groupByOrdinal && a.groupingExpressions.exists(isIntLiteral) =>
```

Comment:
Since you are moving group by to the parser, how about moving both to the parser?